### PR TITLE
Restructuring XML into Table form

### LINKED ISSUE #7901: Restructuring XML into Table form
Similar to the new way of processing JSON to an In-Memory table, we want a simple way to convert an `XML_Document` and `XML_Element` to a table.

Design of what makes sense.

- Name would be a column,
- Namespace should be added if present as a column.
- Attributes would become columns.
- Children becomes a column with `Vector XML_Element`
- Text becomes a column (if multiple concatenate).

Extend the `expand_vector_to_rows` function to explode a `Vector` to a set of rows repeating the other.

If an empty node then no value or children.
```xml
<Library catalog="Fiction" letter="A" />
``` 
or
```xml
<Library catalog="Fiction" letter="A"></library>
```

| Name | @catalog | @letter |
| --- | --- | --- |
| Library | Fiction | A |

If no children and only a text inside becomes.
```xml
<Library catalog="Fiction" letter="A">
    Hello
</Library>
```

| Name | @catalog | @letter | value |
| --- | --- | --- | --- |
| Library | Fiction | A | "Hello"

If children then instead of Value add Children.

```xml
<Library catalog="Fiction" letter="A">
    Hello
    <Book author="An Author">My Book<Book>
    World
    <Book author="Another Author">Your Book<Book>
    <Magazine month="August-2023" series="AutoCar">Cheap Cars For You</Magazine>
</Library>
```

| Name | @catalog | @letter | children |
| --- | --- | --- | --- |
| Library | Fiction | A | ["Hello", Book, "World", Book, Magazine] | 

if then expanded to row:

| Name | @catalog | @letter | children |
| --- | --- | --- | --- |
| Library | Fiction | A | Hello |
| Library | Fiction | A | Book "My Book" |
| Library | Fiction | A | World |
| Library | Fiction | A | Book "Another Book" |
| Library | Fiction | A | Magazine "Cheap Cars For You" |

if then expanded to columns (default prefix is `column name plus a ,`:

| Name | @catalog | @letter | children.Name | children.@author | children.Value | children.@month | children.@series |
| --- | --- | --- | --- | --- | --- | --- | --- |
| Library | Fiction | A | *Nothing* | *Nothing* | Hello | *Nothing* | *Nothing* |
| Library | Fiction | A | Book | An Author | My Book | *Nothing* | *Nothing* |
| Library | Fiction | A | *Nothing* | *Nothing* | World | *Nothing* | *Nothing* |
| Library | Fiction | A | Book | Another Author | Another Book | *Nothing* | *Nothing* |
| Library | Fiction | A | Magazine | *Nothing* | Cheap Cars For You | August-2023 | AutoCar |

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Important Notes

Adds `.to Table` support, as well as XML support for `expand_column`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso [hunk 1]
```diff
@@ -29,6 +29,7 @@ import project.Data.Report_Unmatched.Report_Unmatched
 import project.Data.Row.Row
 import project.Data.Set_Mode.Set_Mode
 import project.Data.Sort_Column.Sort_Column
+import project.Extensions.Table_Conversions
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso:32]: We should import extension modules as:
```
from project.Extensions.Table_Conversions import all
```

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso [hunk 2]
```diff
@@ -1186,8 +1186,8 @@ type Table
           name is used.
     @column Widget_Helpers.make_column_name_selector
     expand_column : Text | Integer -> Vector | Nothing -> Text | Table -> Table ! Type_Error | No_Such_Column | Index_Out_Of_Bounds
-    expand_column self (column : Text | Integer) (fields : Vector | Nothing = Nothing) (prefix : Text | Nothing = Nothing) =
-        Expand_Objects_Helpers.expand_column self column fields prefix
+    expand_column self (column : Text | Integer) (fields : Vector | Nothing = Nothing) (prefix : Text | Nothing = Nothing) (omit_prefix : Boolean = False) =
+        Expand_Objects_Helpers.expand_column self column fields prefix omit_prefix
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso]: we had this option by having empty string as the prefix to omit it.

Maybe an atom type would clearer:
```
type Prefix_Name
     None
     Column_Name
     Custom value:Text
```

**Review by jdunkerley — APPROVED** [review]
Generally, looks good.
I think an Atom type rather than `omit_prefix` would be clearer (and better than the current using empty string).

**Review by radeusgd — APPROVED** [review]
Looks great! 🎉

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso (hunk 2 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso]: Have a `Prefix_Name.from (that:Text)` and then if a user enters text it just works.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso (hunk 1 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso:32]: Done

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso (hunk 2 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso]: Done

**Pull request merged**